packages/update-monitor/src/projects/orbitBridge/contracts/governance.ts

```diff
@@ -0,0 +1,28 @@
+import { providers } from 'ethers'
+
+import { getEip897Implementation } from '../../../common/eip897'
+import { OrbitBridgeProxy__factory } from '../../../typechain'
+import { ContractParameters } from '../../../types'
+import { addresses } from '../constants'
+
+export async function getBridgeWithGovernance(
+  provider: providers.JsonRpcProvider,
+): Promise<ContractParameters> {
+  const bridgeProxy = OrbitBridgeProxy__factory.connect(
+    addresses.bridge,
+    provider,
+  )
+
+  return {
+    name: 'Bridge with Governance',
+    address: addresses.bridge,
+    upgradeability: {
+      type: 'proxy',
+      implementation: await getEip897Implementation(provider, addresses.bridge),
+    },
```

Comment:
I have checked and this bridge does not follow https://eips.ethereum.org/EIPS/eip-897, because it does not have the `proxyType()` function. Maybe `Eip897` should be renamed to `Generic` in our code?

Reply:
I saw that but also noticed that Etherscan calls such proxies (without `proxyType()`) EIP-897 anyway. Of about 5 random EIP-897 proxies on Etherscan I checked, only one had `proxyType()`. So I reluctantly decided to check only for `implementation()` in my code, although it didn't feel quite right. But, since you noticed this as well, I agree we should stick to the EIP requirements. I have updated the code to check for `proxyType()` too.

As for the Orbit Bridge contract it actually gets even more challenging than that. It has a lot of functions already and calls the proxy contract only as a fallback. So it's more of an "extension" mechanism rather than full proxy.